dbms/src/Columns/ColumnString.h

```diff
@@ -250,6 +250,28 @@ class ColumnString final : public COWPtrHelper<IColumn, ColumnString>
         offsets.push_back(new_size);
     }
 
+    template <bool add_terminating_zero>
+    ALWAYS_INLINE inline void batchInsertDataImpl(size_t num, const char * pos, size_t length)
+    {
+        size_t old_size = chars.size();
+        size_t appended_size = 0;
+        const size_t single_str_size = length + (add_terminating_zero ? 1 : 0);
+
+        chars.resize(old_size + (single_str_size * num));
+        if (offsets.capacity() < offsets.size() + num)
+            offsets.reserve(offsets.size() + num);
+
+        for (size_t i = 0; i < num; i++)
+        {
+            inline_memcpy(&chars[old_size + appended_size], pos, length);
+            appended_size += single_str_size;
+
+            if constexpr (add_terminating_zero)
+                chars[old_size + length] = 0;
```

Comment:
I think it should be something like `chars[old_size + appended_size - 1] = 0`, do you have any test to cover this code?

Reply:
> ld be something like `chars[old_size + appended_siz`

These codes is deleted now